src/iexamine.cpp

```diff
@@ -2332,6 +2332,97 @@ void iexamine::kiln_full( player &, const tripoint &examp )
     add_msg( _( "It has finished burning, yielding %d charcoal." ), result.charges );
 }
 
+void iexamine::autoclave_empty( player &p, const tripoint &examp )
+{
+    furn_id cur_autoclave_type = g->m.furn( examp );
+    furn_id next_autoclave_type = f_null;
+    if( cur_autoclave_type == furn_id( "f_autoclave" ) ) {
+        next_autoclave_type = furn_id( "f_autoclave_full" );
+    } else {
+        debugmsg( "Examined furniture has action autoclave_empty, but is of type %s",
+                  g->m.furn( examp ).id().c_str() );
+        return;
+    }
+
+
+    bool water_is_enough = g->u.crafting_inventory().has_charges( "water", 64 );
```

Comment:
This check should be replaced now that you have the `autoclave` requirement defined in json
Something like:
```cpp
reqs.can_make_with_inventory( p.crafting_inventory() );
```